comm/src/main.rs

```diff
@@ -0,0 +1,246 @@
+use std::{
+    fs::File,
+    io::{prelude::*, BufReader, ErrorKind}, process::exit,
+};
+
+use clap::ArgMatches;
+mod cli;
+
+struct Flags {
+    pub one : bool,
+    pub two : bool,
+    pub three : bool
+}
+
+
+fn main() {
+    let matches = cli::create_app().get_matches();
+
+    let flags = Flags::user_gave(&matches);
+
+    let file1: String = match matches.values_of("file_1") {
+        Some(m) => m.map(String::from).collect(),
+        None => "".to_string(),
+    };
+
+    let file2: String = match matches.values_of("file_2") {
+        Some(m) => m.map(String::from).collect(),
+        None => "".to_string(),
+    };
+
+    let mut content_1 : Vec<String> = Vec::new();
+    let mut content_2 : Vec<String> = Vec::new();
+
+    
+    match File::open(&file1) {
+        Ok(file) => {
+            for line in BufReader::new(file).lines() {
+                let line = match line {
+                    Ok(line) => line,
+                    Err(error) => {
+                        eprintln!("comm: {}", error);
+                        exit(1);
+                    },
+                };
+                content_1.push(line);
+            }
+        },
+        Err(e) => {
+            match e.kind() {
+                ErrorKind::NotFound => {
+                    eprintln!("comm: {}: No such file or directory", file1)
+                },
+                ErrorKind::PermissionDenied => {
+                    eprintln!("comm: {}: Permission denied", file1)
+                },
+                _ => eprintln!("comm: {}: Unknown error", file1),
+            }
```

Comment:
There was any particular reason behind not printing the error message of the `io::Error`?

Reply:
I am sorry I left ambiguity there. Fixed it.